authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerAuthorizationHDFSPlugin.java

```diff
@@ -499,27 +506,32 @@ public List<AuthorizationSecurableObject> translatePrivilege(SecurableObject sec
                 case CREATE_SCHEMA:
                   switch (securableObject.type()) {
                     case METALAKE:
+                      NameIdentifier[] catalogs =
+                          GravitinoEnv.getInstance()
+                              .catalogDispatcher()
+                              .listCatalogs(Namespace.of(identifier.name()));
+                      for (NameIdentifier catalog : catalogs) {
+                        AuthorizationUtils.getMetadataObjectLocation(
+                                catalog, Entity.EntityType.CATALOG)
+                            .forEach(
+                                locationPath ->
+                                    createPathBasedMetadataObject(
+                                        securableObject,
+                                        locationPath,
+                                        rangerSecurableObjects,
+                                        rangerPrivileges));
+                      }
+                      break;
```

Comment:
This part is almost the same as L470. Can you try to extract a common method?

Reply:
OK.